data_diff/diff_tables.py

```diff
@@ -123,6 +124,40 @@ def _get_stats(self) -> DiffStats:
 
         return DiffStats(diff_by_sign, table1_count, table2_count, unchanged, diff_percent)
 
+    def _get_stats_dbt(self) -> DiffStats:
+        list(self)  # Consume the iterator into result_list, if we haven't already
+
+        key_columns = self.info_tree.info.tables[0].key_columns
+        diff_by_key = {}
+        extra_column_values_store = {}
+        
+        extra_columns = self.info_tree.info.tables[0].extra_columns
+        extra_column_diffs = {k:0 for k in extra_columns}
+
+        for sign, values in self.result_list:
+            k = values[: len(self.info_tree.info.tables[0].key_columns)]
+            extra_column_values = values[len(self.info_tree.info.tables[0].key_columns) :]
+            if k in diff_by_key:
+                assert sign != diff_by_key[k]
+                diff_by_key[k] = "!"
+                for i in range(0,len(extra_columns)):
```

Comment:
Better to implement this using `zip()` / `safezip()`

Reply:
Thanks for the tip @erezsh that's new to me, I'll try it out.